### PR TITLE
Unit Note Panels

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#15875

*Description of changes:*
* Implements expansion panels for Unit Notes
* Interfaces, enums and resources to support basic unit note structure
* unit tests and mock data
* Current placeholder values to populate unit notes table
* Misc UX for Vue upgrade
* Implements storing and getting of Unit Notes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/unitNotes/UnitNotePanels.vue [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    }
+  },
+  setup () {
+    const router = useRouter()
+
+    const localState = reactive({
+      activePanels: []
+    })
+
+    const getUnText = (noteType: UnitNoteDocTypes): string => {
```
Comment (orelbn) [inline ppr-ui/src/components/unitNotes/UnitNotePanels.vue]: Should this be getUnitNoteText?

File: ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts [hunk 2]
```diff
@@ -163,7 +163,7 @@ export interface MhRegistrationSummaryIF {
   documentRegistrationNumber?: string
   expand?: boolean // used in UI table to toggle expansion.
   expireDays?: number // Number of days until expiry
-  lienRegistrationType?: string
+  lienRegistrationType?: string,
```
Comment (orelbn) [inline ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts]: Did you mean to add the the comma here?

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1375-d6qk4h77.web.app

**Review by orelbn — COMMENTED** [review]
A couple of small comments.
Looks great!

File: ppr-ui/src/components/unitNotes/UnitNotePanels.vue (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/unitNotes/UnitNotePanels.vue]: I dunno, should it? 😆 
Can update if you like!

File: ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts]: Definitely not in an interface, good catch!

File: ppr-ui/src/components/unitNotes/UnitNotePanels.vue (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/unitNotes/UnitNotePanels.vue]: Updated 👍

File: ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts]: Fixed

File: ppr-ui/src/components/unitNotes/UnitNotePanels.vue (hunk 1 above)
Comment (orelbn) [reply ppr-ui/src/components/unitNotes/UnitNotePanels.vue]: Sure! Just to keep with being overly verbose as we usually do

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1375-d6qk4h77.web.app

**Review by orelbn — APPROVED** [review]
Awesome!

File: ppr-ui/src/components/unitNotes/UnitNotePanels.vue (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/unitNotes/UnitNotePanels.vue]: haha can't argue with that.

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1375-d6qk4h77.web.app

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1375-d6qk4h77.web.app

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1375-d6qk4h77.web.app

File: ppr-ui/src/composables/mhrInformation/useMhrInformation.ts [hunk 3]
```diff
@@ -114,6 +115,9 @@ export const useMhrInformation = () => {
     const currentOwnerGroups = data?.ownerGroups || [] // Safety check. Should always have ownerGroups
     await parseMhrHomeOwners(cloneDeep(currentOwnerGroups))
 
+    const unitNotes = data?.notes || []
+    await setMhrUnitNotes(unitNotes)
```
Comment (dimak1) [inline ppr-ui/src/composables/mhrInformation/useMhrInformation.ts:119]: Okkk, now I see what you are doing here. You just parsing the notes from registrations. 👍🏻

**Review by dimak1 — APPROVED** [review]
Great stuff! All good, we can merge this first.

File: ppr-ui/src/composables/mhrInformation/useMhrInformation.ts (hunk 3 above)
Comment (cameron-eyds) [reply ppr-ui/src/composables/mhrInformation/useMhrInformation.ts:119]: Nice! Yessir

**Review by doug-lovett — APPROVED** [review]
Fantastico

File: ppr-ui/src/composables/mhrInformation/useMhrInformation.ts (hunk 3 above)
Comment (cameron-eyds) [reply ppr-ui/src/composables/mhrInformation/useMhrInformation.ts:119]: This is where we could use `mockUnitNotes` until we can get some dummy data in dev.

**Pull request merged**